src/tools/annotation/bidirectionalTool/moveHandle/getBaseData.js

```diff
@@ -0,0 +1,34 @@
+import external from '../../../../externalModules.js';
+import getDistance from './getDistance.js';
+
+export default function getBaseData(data, eventData, fixedPoint) {
+  const { lineSegment } = external.cornerstoneMath;
+  const { start, end, perpendicularStart, perpendicularEnd } = data.handles;
+  const cps = eventData.image.columnPixelSpacing || 1;
+  const rps = eventData.image.rowPixelSpacing || 1;
+
+  const longLine = {
+    start,
+    end,
+  };
+
+  const intersection = lineSegment.intersectLine(longLine, {
+    start: perpendicularStart,
+    end: perpendicularEnd,
+  });
+
+  const distanceToFixed = getDistance(cps, rps, fixedPoint, intersection);
+
+  return {
+    cps,
+    rps,
+    start,
+    end,
+    perpendicularStart,
+    perpendicularEnd,
+    longLine,
+    intersection,
+    distanceToFixed,
+    fixedPoint,
+  };
```

Comment:
I haven't used this tool much. For my own understanding:

```js
  return {
    cps,    // column pixel spacing
    rps,    // row pixel spacing
    start,   // start of the "primary" line?
    end,    // end of the "primary" line?
    perpendicularStart,  // start of "secondary" line?
    perpendicularEnd,   // end of "secondary" line?
    longLine,  // If this is just { start, end }, why return it?
    intersection,  // point where "primary" and "secondary" lines intersect
    distanceToFixed,  // distance to fixed point from the intersection?
    fixedPoint,  // the point opposite the grabbed handle? "Fixed"
  };
```

Reply:
I've added comments to the returned attributes.
Thank you!